Source/DafnyLanguageServer/Workspace/GutterIconAndHoverVerificationDetailsManager.cs

```diff
@@ -309,13 +309,40 @@ public class GutterIconAndHoverVerificationDetailsManager : IGutterIconAndHoverV
           targetMethodNode.StatusVerification = finalOutcome;
         }
 
+        var options = compilation.Program.Reporter.Options;
+        ReportVacuityAndRedundantAssumptionsChecks(compilation, implementation, verificationResult, options);
+
         targetMethodNode.PropagateChildrenErrorsUp();
         targetMethodNode.RecomputeAssertionBatchNodeDiagnostics();
         PublishGutterIcons(compilation, uri, true);
       }
     }
   }
 
+  private static void ReportVacuityAndRedundantAssumptionsChecks(CompilationAfterResolution compilation,
+    Implementation implementation, VerificationResult verificationResult, DafnyOptions options) {
+    if (!options.Get(CommonOptionBag.WarnContradictoryAssumptions)
+        && !options.Get(CommonOptionBag.WarnRedundantAssumptions)
+       ) {
+      return;
+    }
+
+    var vcResults = verificationResult.VCResults.Select(r =>
+      new DafnyConsolePrinter.VCResultLogEntry(r.vcNum, r.startTime, r.runTime, r.outcome,
+        r.asserts.Select(a => (a.tok, a.Description.SuccessDescription)).ToList(), r.coveredElements,
+        r.resourceCount)
+    ).ToList();
+    var res = new DafnyConsolePrinter.VerificationResultLogEntry(
```

Comment:
Something I'd like to do in the future is:

Use CompilationManager for the CLI as well, then the call to `WarnAboutSuspiciousDependencies` in `DafnyDriver` will be subsumed by the call to `WarnAboutSuspiciousDependenciesForImplementation` you have here.

Tweak `VerificationResultLogger` so it uses the `CompilationAfterResolution.ImplementationsPerVerifiable` instead of `DafnyConsolePrinter.VerificationResults` and remove `DafnyConsolePrinter.VerificationResults`

Reply:
Yeah, I'd like to do that in the future, too!